crates/client/rpc/src/events.rs

```diff
@@ -1,10 +1,66 @@
 use std::iter::Skip;
 use std::vec::IntoIter;
 
+use log::error;
+use mc_rpc_core::utils::get_block_by_block_hash;
+use mp_starknet::block::Block;
 use mp_starknet::execution::types::Felt252Wrapper;
+use mp_starknet::traits::hash::HasherT;
+use mp_starknet::traits::ThreadSafeCopy;
 use mp_starknet::transaction::types::EventWrapper;
+use pallet_starknet::runtime_api::{ConvertTransactionRuntimeApi, StarknetRuntimeApi};
+use sc_client_api::backend::{Backend, StorageProvider};
+use sp_api::ProvideRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::traits::Block as BlockT;
+use starknet_core::types::BlockId;
 use starknet_ff::FieldElement;
 
+use crate::errors::StarknetRpcApiError;
+use crate::Starknet;
+
+impl<B, BE, C, P, H> Starknet<B, BE, C, P, H>
+where
+    B: BlockT,
+    C: HeaderBackend<B> + StorageProvider<B, BE> + 'static,
+    C: ProvideRuntimeApi<B>,
+    C::Api: StarknetRuntimeApi<B> + ConvertTransactionRuntimeApi<B>,
+    BE: Backend<B>,
+    H: HasherT + ThreadSafeCopy,
```

Comment:
Are you sure that everything here is absolutely necessary?
Did you try removing some of those

Reply:
Yes, I just checked again as well, everything is necessary